app/models/course_lesson_part.rb

```diff
@@ -3,7 +3,7 @@
 class CourseLessonPart < ApplicationRecord
   belongs_to :course_lesson
 
-  has_one :next_lesson_part, class_name: "CourseLessonPart", foreign_key: :previous_lesson_part_id
+  has_one :next_lesson_part, class_name: "CourseLessonPart", foreign_key: :previous_lesson_part_id, dependent: :destroy
```

Comment:
I am certain we don't want that 😂 That would wipe all the previous lesson parts in sequence.

`dependent : :nullify` is more like it.

I think in the app we were deleting second lesson part, and in tests you were deleting first one. difference is, if you delete first one, the second lesson part still has the id pointing at now gone first part.

Reply:
ahhh ok, yeah i saw nullify as well but thought destroy was getting rid of any reference to the deleted part rather than all the lesson parts.